### ISSUE TITLE
the fine-tuning of animal models

### ISSUE THREAD

**Issue by ZardZen** [KwaiVGI/LivePortrait#506]
Could you provide more details about the fine-tune process, such as the data volume, hyper-parameter configuration, training duration, etc.?

**Comment by EmmaThompson123** [comment]
I have the same question. I want to finetune the animal generator model, my key issue is the animal model still use human dirving video in inference stage, but in training stage, can we use the animal image as driving, or if still use the human image as driving, how can we get the paired source animal image, driving human image and target animal image ?

**Comment by ZardZen** [comment]
> I have the same question. I want to finetune the animal generator model, my key issue is the animal model still use human dirving video in inference stage, but in training stage, can we use the animal image as driving, or if still use the human image as driving, how can we get the paired source animal image, driving human image and target animal image ?

Hi, I just fintuned the generator of stage1, with both human and animal videos. In stage1, I just used 2 frames within a videos as source and driving, no cross-identity generation. I hope it's helpful to you.

**Comment by EmmaThompson123** [comment]
@ZardZen Hi, what about your finetune result ? You say "with both human and animal videos", does this mean you use human and animal videos as both source and driving to train only the animal model ? I also want to use 2 frames within a animal videos as source and driving, but I'm not sure if this will work, because in this scenario, the driving uses animal images during training, but uses human images during inference.  This means there's a gap in the driving used during training and inference.

**Comment by ZardZen** [comment]
> [@ZardZen](https://github.com/ZardZen) Hi, what about your finetune result ? You say "with both human and animal videos", does this mean you use human and animal videos as both source and driving to train only the animal model ? I also want to use 2 frames within a animal videos as source and driving, but I'm not sure if this will work, because in this scenario, the driving uses animal images during training, but uses human images during inference. This means there's a gap in the driving used during training and inference.

I finetuned the animal model to solve the artifacts of mouth, and the result seems ok, my. Yes, both human and animal videos are used in training, as mentioned in the paper " We find our model can generalize well to animals, e.g., cats and dogs, by fine-tuning on a small dataset of animal portraits combined with the original data ". But the loss is different with traing human model. You can find more details in appendix, section D. In stage1, I think the author just trained the model in self-driving  mode, with source and driving image extracted from the same video. So the gap does not seem like a problem. 

https://github.com/user-attachments/assets/3291f498-bb6e-4739-8c8f-13a3d496aa7b

**Comment by EmmaThompson123** [comment]
@ZardZen The result looks so good ! It gave me hope :)  
Could you please provide more details about the fine-tune process, such as the components of the training loss items, the components and amount of the training dataset, and hyper-parameter configuration(e.g. optimizer lr and betas, crop scale) etc ? Thanks anyway !

**Comment by ZardZen** [comment]
> [@ZardZen](https://github.com/ZardZen) The result looks so good ! It gave me hope :) Could you please provide more details about the fine-tune process, such as the components of the training loss items, the components and amount of the training dataset, and hyper-parameter configuration(e.g. optimizer lr and betas, crop scale) etc ? Thanks anyway !

@EmmaThompson123  Yes, and I am glad that it is helpful. I basically followed the training settings of stage 1 in the paper, with a lr of 2e-4, beta1 0.5, and beta2 0.999. The losses include keypoints equivariance loss, keypoint prior loss, and deformation prior loss, landmark-guided loss, also perceptual and GAN loss. The head pose loss, faceid loss and lip GAN loss is unused according to section D of appendix. Training dataset compose of 2 parts of human and animal, but I don't have too much data of animals. Around 1-2k animal videos. The crop scale I used is from the inference pipeline of official codes, I can't remember if it's 2.3 or 2.5. I did this work about half years ago, some details may not be very clear,  I hope it could be helpful!

**Comment by EmmaThompson123** [comment]
@ZardZen Thanks for your reply ! I have a question: In stage 1, if we only train the generator, do we still need the keypoints equivariance loss, keypoint prior loss, and deformation prior loss, landmark-guided loss, I think these losses are used for training the motion extractor. BTW, what is the data volume of human training data ? Thanks again !

**Comment by ZardZen** [comment]
> [@ZardZen](https://github.com/ZardZen) Thanks for your reply ! I have a question: In stage 1, if we only train the generator, do we still need the keypoints equivariance loss, keypoint prior loss, and deformation prior loss, landmark-guided loss, I think these losses are used for training the motion extractor. BTW, what is the data volume of human training data ? Thanks again !

@EmmaThompson123 Yes, if you don't fine-tune other modules, these losses are not necessary. I fine-tuned all modules in stage 1, so they were used during training process. You could just freeze other modules except decoder module, without these losses, I think it would work too. The amount of human training data  is about 10k videos, but in dataloader, human data were sampled less times than animal data. I don't have too much data, neither human nor animal data. I think more data you use for training, the fine-tune result would be much better. Best wishes !

**Comment by EmmaThompson123** [comment]
@ZardZen Hello, I have a question about how the Landmark-guided loss is calculated in the pet version of LivePortrait. I've seen other issues mentioning that 10 landmarks are selected for calculation, but that's for human faces, not pet faces (which only have 9 landmarks if use x-pose). Also, the order of these selected landmarks should correspond to the semantic positions of the 21 implicit landmarks in the pre-trained model, right? For example, if the first implicit landmark is located at the center of the right eye, then the first guided landmark should also be at the center of the right eye. So, how do you calculate the Landmark-guided loss in the pet version of LivePortrait? Specifically, this question includes: which explicit landmark detection model are you using, and which of the explicit landmarks correspond to which of the implicit landmarks?

**Comment by leebastien60-design** [comment]
@ZardZen Hello, could you please open source your code or model ~
